contracts/marketplace/src/multitest.rs

```diff
@@ -3761,3 +3764,92 @@ fn listing_funds(listing_fee: u128) -> Result<Vec<Coin>, ContractError> {
         Ok(vec![])
     }
 }
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+// SudoParamsV015 represents the previous state from v0.15.0 version
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct SudoParamsV015 {
+    pub trading_fee_percent: Decimal,
+    pub ask_expiry: ExpiryRange,
+    pub bid_expiry: ExpiryRange,
+    pub operators: Vec<Addr>,
+    pub max_finders_fee_percent: Decimal,
+    pub min_price: Uint128,
+    pub stale_bid_duration: Duration,
+    pub bid_removal_reward_percent: Decimal,
+}
+
+use cw2::set_contract_version;
+use cw_storage_plus::Item;
+#[test]
+fn try_migrate() {
+    let mut deps = mock_dependencies();
+    let env = mock_env();
+
+    let old_params_item: Item<SudoParamsV015> = Item::new("sudo-params");
+
+    let old_params = SudoParamsV015 {
+        operators: vec![Addr::unchecked("operator1")],
+        trading_fee_percent: Decimal::percent(TRADING_FEE_BPS),
+        ask_expiry: ExpiryRange::new(MIN_EXPIRY, MAX_EXPIRY),
+        bid_expiry: ExpiryRange::new(MIN_EXPIRY, MAX_EXPIRY),
+        max_finders_fee_percent: Decimal::percent(MAX_FINDERS_FEE_BPS),
+        min_price: Uint128::from(5u128),
+        stale_bid_duration: Duration::Time(100),
+        bid_removal_reward_percent: Decimal::percent(BID_REMOVAL_REWARD_BPS),
+    };
+    old_params_item
+        .save(&mut deps.storage, &old_params)
+        .unwrap();
+
+    // should error when different name
+    set_contract_version(&mut deps.storage, "crates.io:marketplace", "0.15.0").unwrap();
+    let err = migrate(deps.as_mut(), env.clone(), Empty {}).unwrap_err();
+
+    assert_eq!(
+        err.to_string(),
+        "Generic error: Cannot upgrade to a different contract"
+    );
+
+    // should error when version is greater version
+    set_contract_version(&mut deps.storage, "crates.io:sg-marketplace", "2.0.0").unwrap();
+    let err = migrate(deps.as_mut(), env.clone(), Empty {}).unwrap_err();
+    assert_eq!(
+        err.to_string(),
+        "Generic error: Cannot upgrade to a previous contract version"
+    );
+
+    // no op when same version
+    set_contract_version(
+        &mut deps.storage,
+        "crates.io:sg-marketplace",
+        env!("CARGO_PKG_VERSION"),
+    )
+    .unwrap();
+    migrate(deps.as_mut(), env.clone(), Empty {}).unwrap();
+
+    set_contract_version(&mut deps.storage, "crates.io:sg-marketplace", "0.15.0").unwrap();
+    migrate(deps.as_mut(), env, Empty {}).unwrap();
+
+    let new_params = SUDO_PARAMS.load(&deps.storage).unwrap();
+
+    assert_eq!(new_params.operators, old_params.operators);
```

Comment:
I like https://github.com/google/assertor and `assert_that!(new_params.operators).is_equal_to(old_params.operators)` which gives better error reporting when it fails.